compiler/src/dotty/tools/dotc/transform/LambdaLift.scala

```diff
@@ -444,6 +444,7 @@ object LambdaLift {
 
     def liftDef(tree: MemberDef)(implicit ctx: Context): Tree = {
       val buf = liftedDefs(tree.symbol.owner)
+      tree.symbol.setFlag(Lifted)
```

Comment:
It would be better to set the flag when the denotation is transformed in liftLocals: https://github.com/Duhemm/dotty/blob/ee2d91e5e93be553751a2156e911540a66afb1da/compiler/src/dotty/tools/dotc/transform/LambdaLift.scala#L344